File: frontend/src/app/core/interface/response/post-response.ts
export interface PostResponse {
    id?: string;
    image?:string;
    // image?:File;
    title?: string;
    content?: string;
    tags?: string[];
    createdDate?: Date | string;
    createdBy?: string;
    lastModifiedDate?: Date | string;
    lastModifiedBy?: string;
}


// export interface PostResponse {
//     id: string;
//     image?: string;
//     title: string;
//     content: string;
//     tags: string[];
//     createdDate: Date;
//     createdBy: string;
//     lastModifiedDate: Date;
//     lastModifiedBy: string;
//   }
  

File: backend/src/main/java/com/tuankhoi/backend/dto/response/PostResponse.java
package com.tuankhoi.backend.dto.response;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class PostResponse implements Serializable {
    String id;

    String title;

    String description;

    String content;

    String subCategoryId;

    Set<String> tags;

    String authorId;

    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    LocalDateTime createdDate;

    String lastModifiedBy;

    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    LocalDateTime lastModifiedDate;

    int likeCount;

    int commentCount;

    int viewCount;
}
